src/OrchardCore.Modules/OrchardCore.Users/Workflows/Activities/ResetUserPasswordTask.cs

```diff
@@ -0,0 +1,155 @@
+        public override string Name => nameof(ResetUserPasswordTask);
+
+        public override LocalizedString DisplayText => S["Reset User Password Task"];
+
+        // The category to which this activity belongs. The activity picker groups activities by this category.
+        public override LocalizedString Category => S["User"];
+
+        public WorkflowExpression<string> UserName
+        {
+            get => GetProperty(() => new WorkflowExpression<string>());
+            set => SetProperty(value);
+        }
+
+        public WorkflowExpression<string> ResetPasswordEmailSubject
+        {
+            get => GetProperty(() => new WorkflowExpression<string>());
+            set => SetProperty(value);
+        }
+
+        // The message to display.
+        public WorkflowExpression<string> ResetPasswordEmailTemplate
+        {
+            get => GetProperty(() => new WorkflowExpression<string>());
+            set => SetProperty(value);
+        }
+
+        // Returns the possible outcomes of this activity.
+        public override IEnumerable<Outcome> GetPossibleOutcomes(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            return Outcomes(S["Done"], S["Invalid"]);
+        }
+
+        // This is the heart of the activity and actually performs the work to be done.
+        public override async Task<ActivityExecutionResult> ExecuteAsync(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            var userName = await _expressionEvaluator.EvaluateAsync(UserName, workflowContext);
+
+            var user = await _userService.GetForgotPasswordUserAsync(userName) as User;
+            if (user == null || ((await _siteService.GetSiteSettingsAsync()).As<RegistrationSettings>().UsersMustValidateEmail && !await _userManager.IsEmailConfirmedAsync(user)))
+            {
+                var updater = _updateModelAccessor.ModelUpdater;
+                if (updater != null)
+                {
+                    updater.ModelState.TryAddModelError("", S["No email service is available"]);
+                }
+                return Outcomes("Invalid");
+            }
+            else
+            {
+                user.ResetToken = Convert.ToBase64String(Encoding.UTF8.GetBytes(user.ResetToken));
+                var resetPasswordUrl = _linkGenerator.GetUriByAction(_httpContextAccessor.HttpContext,
+                    "ResetPassword", "ResetPassword", new { area = "OrchardCore.Users", code = user.ResetToken });
+
+                workflowContext.Properties["ResetPasswordUrl"] = resetPasswordUrl;
+
+                var subject = await _expressionEvaluator.EvaluateAsync(ResetPasswordEmailSubject, workflowContext);
+                var localizedSubject = new LocalizedString(nameof(ResetUserPasswordTask), subject);
+
+                var body = await _expressionEvaluator.EvaluateAsync(ResetPasswordEmailTemplate, workflowContext);
```

Comment:
var body = _expressionEvaluator.EvaluateAsync(Body, workflowContext, _htmlEncoder);

Reply:
Thanks @Skrypt ! I was on an older version of OC so didn't notice this failed build when merging in